security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -1042,10 +1041,16 @@ func (sc *SecretCache) getExchangedToken(ctx context.Context, k8sJwtToken string
 	return exchangedTokens[0], nil
 }
 
-func (sc *SecretCache) certExists() bool {
+// useCertToRotate checks if we can use cert instead of token to do CSR.
+func (sc *SecretCache) useCertToRotate() bool {
```

Comment:
I don't think it's good to hardcode "GoogleCA" cases all over the code. 
Add an explicit setting 'alwaysUseToken' - or even better, check if the token is JWT and not expired and send it along in all requests. It is fine to have both JWT and to set the certificate in the request - GoogleCA will use the JWT and ignore the certs.

Reply:
Adding a "alwaysUseToken" flag sounds good. Attaching token and cert sounds appealing, but what if token expired, should we try to get a new token? what if fetching token failed? should we fail directly or try just using cert for CSR? The error handling becomes complicated if we do not know what is the intended behavior.

Reply:
Added UseTokenForCSR flag.